table/queries/merged_reduced_satellite_scans.sql

```diff
@@ -0,0 +1,142 @@
+    WHEN REGEXP_CONTAINS(error, '"Err": 90') THEN "read/dns.msgsize"
+    WHEN REGEXP_CONTAINS(error, '"Err": 111') THEN "read/udp.refused"
+    WHEN REGEXP_CONTAINS(error, '"Err": 113') THEN "read/ip.host_no_route"
+    WHEN error = "{}" THEN "dns/unknown"
+    WHEN error = "no_answer" THEN "dns/dns.name_not_resolved"
+    #Satellite v2
+    WHEN ENDS_WITH(error, "i/o timeout") THEN "read/udp.timeout"
+    WHEN ENDS_WITH(error, "message too long") THEN "read/dns.msgsize"
+    WHEN ENDS_WITH(error, "connection refused") THEN "read/udp.refused"
+    WHEN ENDS_WITH(error, "no route to host") THEN "read/ip.host_no_route"
+    WHEN ENDS_WITH(error, "short read") THEN "read/dns.msgsize"
+    ELSE "dns/unknown"
+  END
+);
+
+# Input: array of received IP information, array of rcodes, error,
+#        controls failed, and anomaly fields from the raw data
+# Output is a string of the format "stage/outcome"
+CREATE TEMP FUNCTION SatelliteOutcome(received ANY TYPE, rcode ARRAY<STRING>, error STRING, controls_failed BOOL, anomaly BOOL) AS (
+  CASE
+    WHEN controls_failed THEN "setup/controls"
+    WHEN ARRAY_LENGTH(received) > 0 THEN
+      CASE
+        WHEN anomaly THEN "dns/dns.ipmismatch"
+        ELSE "expected/match"
+      END
+    WHEN ARRAY_LENGTH(rcode) > 0 THEN ClassifySatelliteError(rcode[ORDINAL(ARRAY_LENGTH(rcode))], SPLIT(error, " | ")[ORDINAL(ARRAY_LENGTH(SPLIT(error, " | ")))])
+    ELSE ClassifySatelliteError("", SPLIT(error, " | ")[ORDINAL(ARRAY_LENGTH(SPLIT(error, " | ")))])
+  END
+);
+
+# BASE_DATASET and DERIVED_DATASET are reserved dataset placeholder names
+# which will be replaced when running the query
+
+# Increment the version of this table if you change the table in a backwards-incomatible way.
+
+# Rely on the table name firehook-censoredplanet.derived.merged_reduced_scans_vN
+# if you would like to see a clear breakage when there's a backwards-incompatible change.
+# Old table versions will be deleted.
+CREATE OR REPLACE TABLE `firehook-censoredplanet.DERIVED_DATASET.merged_reduced_satellite_scans_v2`
+PARTITION BY date
+# Columns `source` and `country_name` are always used for filtering and must come first.
+# `network` and `domain` are useful for filtering and grouping.
+CLUSTER BY source, country_name, network, domain
+OPTIONS (
+  friendly_name="Reduced Satellite Scans",
+  description="Filtered and pre-aggregated table of Satellite scans to use with the Censored Planed Dashboard"
+)
+AS (
+WITH Grouped AS (
+    SELECT
+        date,
+
+        source,
+        country AS country_code,
+        as_full_name AS network,
+        IF(is_control, "CONTROL", domain) AS domain,
+
+        SatelliteOutcome(received, rcode, error, controls_failed, anomaly) as outcome
+        CONCAT("AS", asn, IF(organization IS NOT NULL, CONCAT(" - ", organization), "")) AS subnetwork,
+        IFNULL(category, "Uncategorized") AS category,
+
+        COUNT(1) AS count
+    FROM `firehook-censoredplanet.BASE_DATASET.satellite_scan`
```

Comment:
once we deploy code with this query we'll need to be sure this base table doesn't get deleted while we're changing it, or the nightly pipeline could fail at the end.